Sources/Apollo/GraphQLExecutor.swift

```diff
@@ -125,20 +124,30 @@ final class GraphQLExecutor {
                                                       withKey key: CacheKey? = nil,
                                                       variables: GraphQLMap? = nil,
                                                       accumulator: Accumulator) throws -> Promise<Accumulator.FinalResult> {
-    let info = GraphQLResolveInfo(rootKey: key, variables: variables)
-
-    return try execute(selections: selections,
-                       on: object,
-                       info: info,
-                       accumulator: accumulator).map {
-      try accumulator.finish(rootValue: $0, info: info)
-    }.asPromise()
+    if #available(OSX 10.12, iOS 10.0, tvOS 10.0, watchOS 3.0, *) {
+      dispatchPrecondition(condition: .notOnQueue(queue))
```

Comment:
Yep, good point. `dispatchPreconditionIfAvailable`?

Reply:
Yeah looks like I'm using `IfPossible` as the suffix [on a similar wrapper method here](https://github.com/apollographql/apollo-ios/blob/main/Sources/Apollo/JSONSerialization%2BSorting.swift#L10), but `ifAvailable` works too

Reply:
Hmmm, it turns out this is harder than I thought, because the `DispatchPredicate` type (what is passed as `condition`) is also unavailable on older OS versions. So in order to define a wrapper function we'd have to define our own `DispatchPredicate` type and translate those values to the actual type. That seems like a bigger mess than leaving the availability conditions in there for now. (Any idea when we would be able to drop support for iOS 9?)

Reply:
We could also just do `checkNotOnQueueIfPossible` for now. 

And I'm going to do a new developer survey soon to try and figure out how many people are still using 9/10.